src/cpp/utils/SystemInfo.cpp

```diff
@@ -83,4 +86,35 @@ bool SystemInfo::file_exists(
     return (stat(filename.c_str(), &s) == 0 && s.st_mode & S_IFREG);
 }
 
+ReturnCode_t SystemInfo::load_environment_file(
+        const std::string& filename,
+        const std::string& env_name,
+        std::string& env_value)
+{
+    // Check that the file exists
+    if (!SystemInfo::file_exists(filename))
+    {
+        return ReturnCode_t::RETCODE_BAD_PARAMETER;
+    }
+
+    // Read json file
+    std::ifstream file(filename);
+    nlohmann::json file_content;
+    file >> file_content;
+
+    try
+    {
+        env_value = file_content.at(env_name);
+        if (env_value.empty() || env_value.compare("") == 0)
+        {
+            return ReturnCode_t::RETCODE_NO_DATA;
+        }
+    }
+    catch(const std::exception&)
```

Comment:
I would capture only `json::exception`, and let all other exception pass through.

Reply:
addressed in f67682c